e2e/cypress/integration/messaging/inline_images_open_preview_window_spec.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login as test user and visit the newly created test channel
+        cy.apiInitSetup({loginAfter: true}).then(({team, channel}) => {
+            // # Visit a test channel
+            cy.visit(`/${team.name}/channels/${channel.name}`);
+        });
+    });
+    it('M23345 - Inline markdown images open preview window', () => {
+        // # Enable 'Show markdown preview option in message input box' setting in Account Settings > Advanced
+        cy.get('#sidebarHeaderDropdownButton').click();
+        cy.get('#accountSettings').click();
+        cy.get('#advancedButton').click();
+        cy.get('#advancedPreviewFeaturesDesc').click();
+        cy.get('#advancedPreviewFeaturesmarkdown_preview').check().should('be.checked');
+        cy.get('#saveSetting').click();
+        cy.reload();
```

Comment:
This section has a model newly opened. I'd suggest to use `within` to scope locating element within the modal. Also, since it's newly opened, a visibility should be done first.  Lastly, avoid using reload as much as possible since it costly to reload a page and we're also expecting realtime change on user settings. So, this could be,
```suggestion
        cy.get('#accountSettingsModal').should('be.visible').within(() => {
            cy.findByText('Advanced').click();
            cy.findByText('Preview Pre-release Features').should('be.visible').click();
            cy.get('#advancedPreviewFeaturesmarkdown_preview').check().should('be.checked');
            cy.findByText('Save').click();
            cy.findAllByLabelText('Close').first().click();
        });
```

Reply:
Changes made. And makes sense.